cmd/testperformance.go

```diff
@@ -187,102 +217,241 @@ func testPerformance(ctx context.Context, queuedTests []testCaseName, allTests m
 	}
 }
 
-func performanceDiskWriteTest(ctx context.Context, conf *testPerformanceConfig) testResult {
-	testRes := testResult{Name: "DiskWrite"}
-
-	log.Info(ctx, "Testing disk write...",
-		z.Any("file size MB", conf.DiskWriteMB),
-		z.Any("loops", diskWriteLoops))
-
-	if conf.DiskWriteMB <= 3072 {
-		log.Warn(ctx, "File size used for tests of 3072MB or lower impacts the measured performance", nil)
+func fioCommand(ctx context.Context, filename string, fileSize int, blocksize int, operation string) ([]byte, error) {
+	//nolint:gosec
+	cmd, err := exec.CommandContext(ctx, "fio",
+		"--name=fioTest",
+		fmt.Sprintf("--filename=%v", filename),
+		fmt.Sprintf("--size=%vMb", fileSize),
+		fmt.Sprintf("--blocksize=%vk", blocksize),
+		fmt.Sprintf("--numjobs=%v", diskOpsNumOfJobs),
+		fmt.Sprintf("--rw=%v", operation),
+		"--direct=1",
+		"--runtime=60s",
+		"--group_reporting",
+		"--output-format=json",
+	).Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "exec fio command")
 	}
 
+	return cmd, nil
+}
+
+func setupDiskOperation(ctx context.Context, conf *testPerformanceConfig) (int, string, error) {
 	var stat unix.Statfs_t
-	wd, _ := os.UserHomeDir()
-	err := unix.Statfs(wd, &stat)
+	var dir string
+	var err error
+	if conf.DiskIOTestFileDir == "" {
+		dir, err = os.UserHomeDir()
+		if err != nil {
+			return 0, "", errors.Wrap(err, "get user home dir")
+		}
+	} else {
+		dir = conf.DiskIOTestFileDir
+	}
+	err = unix.Statfs(dir, &stat)
```

Comment:
Calling `unix` here effectively makes Charon *NIX only.

Can we avoid this?

Reply:
Hmmm, I thought it is already 😄. I will think about another approach